index.d.ts

```diff
@@ -17,7 +17,7 @@ declare module 'connected-react-router' {
     action: 'POP' | 'PUSH'
   }
 
-  export const LOCATION_CHANGE: string
+  export const LOCATION_CHANGE: '@@router/LOCATION_CHANGE'
```

Comment:
Why do we need to change this? How does it affect the result?

Reply:
This is needed for TypeScript's control flow analysis, e.g.:

``` ts
const reducer = (state, action: LocationChangeAction | MyOtherAction) => {
  switch (action.type) {
    case LOCATION_CHANGE: {
      // Here, TypeScript knows `action` is type `LocationChangeAction`, and has a property called `payload`, which has a property called `location`
    }
  }
}
```

If the action type was not a string literal, TypeScript's control flow analysis wouldn't be able to narrow the union.